lib/snapshot.go

```diff
@@ -443,18 +444,29 @@ func NewSnapshot(
 	//   Either way, it means our snapshot was compromised and we need to recompute it as described
 	//   in the previous bullet.
 	shouldRestart := false
+	isChecksumIssue := false
+	// @diamondhands - I think the only operations tracked by the state semaphore are (1) checksum add, (2) checksum
+	// remove, (3) ProcessChunk, (4) ChecksumPrint, (5) Exit. So the only one operation that isn't checksum related
+	// is the ProcessChunk operation. However, I think we can get away with this being considered a "checksum" issue
+	// and not restarting the node (which causes a panic). If you have a node that didn't finish processing chunks,
+	// just start over again.
 	if operationChannel.StateSemaphore > 0 {
 		operationChannel.StateSemaphore = 0
 		glog.Errorf(CLog(Red, fmt.Sprintf("NewSnapshot: Node didn't shut down properly last time. Entering a "+
 			"recovery mode. The node will roll back to last snapshot epoch block height (%v) and hash (%v), then restart.",
 			metadata.SnapshotBlockHeight, metadata.CurrentEpochBlockHash)))
 		shouldRestart = true
+		isChecksumIssue = true
 	}
 
 	if !shouldRestart {
 		if err := migrations.StartMigrations(); err != nil {
+			// @diamondhands - If we hit this error, it means that the migration failed. Should we
+			// consider this a "checksum" issue or not? I believe the migrations are only necessary
+			// for checksums, but I'm unsure.
```

Comment:
Yep, migrations are only needed to guarantee that checksums match. They don't change anything about how the data the node is processing is handled. I'd add that to the comment and run with it.